src/components/swapv2/ConfirmSwapModal.tsx

```diff
@@ -111,6 +120,10 @@ export default function ConfirmSwapModal({
     [onDismiss, modalBottom, modalHeader, swapErrorMessage],
   )
 
+  useEffect(() => {
+    trade?.encodeSolana()
```

Comment:
dont like this way. it change attribute of `trade`, not change react state, maybe will cause some unexpected error

Reply:
This way is better to return new instance of trade with encoded solana txs. Many effect depend on `trade` may cause unnecessary re-run